Add debian/ dir & build packages in CI (for Debian and derivatives)

tenzap opened the pull request:
Keep in mind that:
- when you update/set the version for a release, it has to be updated in debian/changelog as well. However, the github actions workflow will set it again to build the deb. Maybe you could set the version string to the value of your tag instead of using `${{ github.run_number }} + 121`
- ~~when you update the flutter version you should also update the version in `debian/rules`. The build will fail by design if it is not the same version as the one in your github actions job.~~ With the update to  2.12.5, the version is fetched from the github actions workflow. So no additional maintenance is required here.

~~For now (2.12.4) I have disabled the tests because one test fails. It can be reenabled once the tests PASS.~~ → Tests pass with 2.12.5, so they are enabled again.

Note that this cannot stricly follow the "Debian policy" since (among other reasons), to build the binary package, we have to download flutter SDK from the internet as well as rustup for older distributions.

As you can see for [2.12.4+178-1](https://github.com/tenzap/BloomeeTunes/actions/runs/18092216743) and for [2.12.5+183-1](https://github.com/tenzap/BloomeeTunes/actions/runs/18493145415), the github actions job produces successfully the debs for 6 suites:
- Debian 11 bullseye
- Debian 12 bookworm
- Debian 13 trixie
- Ubuntu 22.04 jammy
- Ubuntu 24.04 noble
- Kali rolling

I can confirm that it runs perfectly on Debian bookworm. I can't test the other, but they should work as well.

HemantKArya commented:
Hi @tenzap, thanks for this commit, Actually I was just busy with some work so unable to get time for review this but as soon as I will review it then I will definitely merge it. 
btw I have a question that is it necessary to build that much different distributions for every different flavor of Linux since there are some other projects who only create simple two files `.deb` or `.rpm`? I am just curious cause never came across in Linux that much.
You can also see example screenshot from VSCode official site too - until then I will review it 😊
<img width="456" height="460" alt="image" src="https://github.com/user-attachments/assets/10d9c247-b383-4f1e-b57d-4ff807196e28" />

tenzap committed:
Add debian/ dir & build packages in CI (for Debian and derivatives)

Keep in mind that:
- when you update/set the version for a release, it has to be updated in debian/changelog as well. However, the github actions workflow will set it again to build the deb.
- when you update the flutter version you should also update the version in debian/rules. The build will fail by design if it is not the same version as the one in your github actions job.

For now I have disabled the tests because one test fails. It can be reenabled once the tests PASS.

Note that this cannot stricly follow the "Debian policy" since (among other reasons), to build the binary package, we have to download flutter SDK from the internet as well as rustup for older distributions.

The github actions job produces the binary packages (*.deb) for 6 suites:
- Debian 11 bullseye
- Debian 12 bookworm
- Debian 13 trixie
- Ubuntu 22.04 jammy
- Ubuntu 24.04 noble
- Kali rolling

I can confirm that it runs perfectly on Debian bookworm. I can't test the other, but they should work as well.

tenzap committed:
github workflow: Add job 'build-debs'

tenzap commented:
I updated this for v3.0.2
Please consider merging. This would solve #65 and the other mentioned issues.

tenzap commented:
@HemantKArya see previous message. What is blocking the merge?
